ci/single_binary_test.d

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env rdmd
+module system_integration_test;
+
+private:
+
+import std.algorithm;
+import std.file;
+import std.format;
+import std.path;
+import std.process;
+import std.stdio;
+import core.thread;
+
+/// Root of the repository
+immutable RootPath = __FILE_FULL_PATH__.dirName.dirName;
+immutable IntegrationPath = RootPath.buildPath("tests").buildPath("system");
+immutable LauncherPath = RootPath.buildPath("tests").buildPath("launcher");
+
+/+ ***************************** Commands to run **************************** +/
+/// A simple test to ensure that the container works correctly,
+/// e.g. all dependencies are installed and the binary isn't corrupt.
+immutable BuildBinary = [ "dub", "build", "-c", "launcher", ];
+immutable BinaryUp = [ "build/agora-launcher", "-c", LauncherPath.buildPath("configs.yaml"), ];
+immutable BinaryDown = [ "killall", "-9", "agora-launcher", ];
+immutable RunIntegrationTests = [ "dub", "--root", IntegrationPath, ];
+immutable Cleanup = [ "rm", "-rf", LauncherPath.buildPath("node/0/.cache/"),
+                      LauncherPath.buildPath("node/1/.cache/"),
+                      LauncherPath.buildPath("node/2/.cache/")];
+
+private int main (string[] args)
+{
+    runCmd(BinaryDown, false);
+
+    runCmd(BuildBinary);
+
+    // First make sure that there we start from a clean slate,
+    // as the docker-compose bind volumes
+    runCmd(Cleanup);
+    Thread.sleep(1.seconds);
+
+    // Now run the tests
+    runCmd(BinaryUp, false);
+    Thread.sleep(2.seconds);
+    scope (exit) runCmd(BinaryDown, false);
+    runCmd(RunIntegrationTests);
+
+    return 0;
+}
+
+/// Utility function to run a command and throw on error
+private void runCmd (const string[] cmd, bool isWait = true)
+{
+    writeln(cmd);
+    auto pid = spawnProcess(cmd);
+    if (isWait)
```

Comment:
You should always wait for a command to finish. If not here, at least return the Pid and do it in the main